api/app/Http/Controllers/BaseController.php

```diff
@@ -1,39 +1,67 @@
 <?php namespace App\Http\Controllers;
 
+use App\Http\Transformers\BaseTransformer;
 use Laravel\Lumen\Routing\Controller;
 use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Illuminate\Http\Request;
+// use Illuminate\Container\Container as App;
 
 class BaseController extends Controller
 {
-
     public static $model;
 
+    /**
+     * Transformer for the response arrays.
+     *
+     * @var App\Services\Transformer
+     */
+    protected $transformer;
+
+    /**
+     * Model Repository.
+     *
+     * @var App\Repositories\BaseRepository
+     */
+    protected $repository;
+
+    /**
+     * Validation service.
+     *
+     * @var App\Services\Validation\TestValidator
+     */
+    protected $validator;
+
+    /**
+     * Get all entities.
+     *
+     * @return Response
+     */
     public function getAll()
     {
-        $model = static::$model;
-
-        $entities = $model::all();
-
-        return response()->json($entities);
-
-//        $this->transformer = $entity.'Transformer';
-//        return $this->response->collection($entities, new $this->transformer);
+        return $this->transformer->collection($this->repository->all(), new BaseTransformer);
```

Comment:
Should the BaseController have a constructor so that the `$this->transformer` is the same as `new BaseTransformer`? There seems to be a discrepancy between the implementation of `BaseController` and `TestController` below